### ISSUE TITLE
Added helper method to close resources after db operation

### ISSUE THREAD

**Issue by DraganD-Ukr** [DraganD-Ukr/Web-Patterns-CA1#4]
I added a **method** that closes _ResultSet_, _PreparedStatement_ and _Connection_ objects. Since we duplicate this piece of code i want to merge it immediately, so you guys use it.

**Comment by Aloy-Lockness11** [comment]
1)Is there a way of using **try with resources** instead of using another method in DAO just for the  _ResultSet_ and _PreparedStatement_?

2)If it is required then isn't it better to have the garbage collection in free connection itself rather than extrapolating it out to another method ? 

` public void freeConnection(Connection con){}`

**Comment by DraganD-Ukr** [comment]
@Aloy-Lockness11 

The helper method i meant: 
`/**
     * Helper method to close resources after database operation
     * @param rs - ResultSet object to be closed
     * @param ps - PreparedStatement object to be closed
     * @param con - Connection object to be closed
     */
    public void closeResources(ResultSet rs, PreparedStatement ps, Connection con) {
        try {
            if (rs != null) {
                rs.close();
            }
            if (ps != null) {
                ps.close();
            }
            if (con != null) {
                freeConnection(con);
            }
        } catch (SQLException e) {
            System.out.println("Exception occurred while closing resources: " + e.getMessage());
        }
    }`

As i understood, we can try-with-resources because they implement AutoCloseable which automatically closes them at the end of the block.
Basically what i did is just extracted the code that **Michelle** uses in here DAOs to Parent 'DAO' class. 

I'm up for try-with-resources solution as it simplifies code and resource management

Just confirm it and i'm gonna implement try with resources across my part of code!

**Comment by DraganD-Ukr** [comment]
@Aloy-Lockness11 

I assume **Michelle** is closing resources if they are **not** **automatically** closed. That's what that part of code does

**Comment by Aloy-Lockness11** [comment]
Rogger understood

**Comment by DraganD-Ukr** [comment]
@Aloy-Lockness11 
So, should we keep it or depend on Autoclose?

**Comment by Aloy-Lockness11** [comment]
I have already discussed with the lecturer and she recommended us to use try with resources as it goes with standards.
I have also come across this thread as well may help us decide an agreement

https://stackoverflow.com/questions/28197155/is-try-catch-more-or-less-expensive-than-try-with-resources

**Comment by DraganD-Ukr** [comment]
Than let's stick with try-with-resources

**Comment by Aloy-Lockness11** [comment]
agreed

**Comment by DraganD-Ukr** [comment]
agreed on try-with-resources
